tests/array_test.py

```diff
@@ -317,7 +317,7 @@ def test_wrong_num_arrays(self):
     inp_data = np.arange(math.prod(shape), dtype=np.float32).reshape(shape)
     di_map = s.devices_indices_map(shape)
     bufs = [jax.device_put(inp_data[di_map[d]], d)
-            for d in jax.local_devices()]
+            for d in jax.local_devices()[:8]]
```

Comment:
Please add a comment wherever you are doing this pattern.

Otherwise it is difficult to know why this happens and I/someone might revert to the normal thing.

Reply:
I think the pattern is obvious enough, though: the device has more than 8 cores, which is simply something we never test ourselves.

Reply:
Sure, but a comment doesn't hurt and the pattern is only in a couple of places.

Reply:
Note that we do run some of these tests on v3-32 where these things might fail. So we need to be careful about that?

Reply:
By the way, if we changed our *CPU* tests to use more devices (say, 16 or something), we'd catch this bug also.

Reply:
Yeah but that's single host.

The v3-32 tests are multihost so it will break with error saying that host2 doesn't have anything to do.

Maybe the tests on IPU can set the devices to 8 (when running jax tests) so these changes are not required?

Reply:
This did seem to be a common pattern in existing tests
https://github.com/google/jax/blob/8acf597ebac658fdfe12580ecedfcc37f5e3cbe1/tests/shard_map_test.py#L307
I've added a small comment to these changes to make it clear it's intentional.